### ISSUE TITLE
Add bubble map support for plotting

### ISSUE THREAD

**Issue by chinandrew** [cmu-delphi/covidcast#43]
Currently only choropleths are supported with plotting. Add support for bubble maps like what is available in the [R package](https://cmu-delphi.github.io/covidcast/covidcastR/articles/plotting-signals.html#bubble-maps)

**Comment by chinandrew** [comment]
Few questions on this one:
1. Do we want points to scale by area or radius? i.e. for a 2x increase, does the radius or area increase by 2x? I figured doing area would be most accurate, but the R package (screenshot) looks like it scales by radius based on my super scientific inspection below. matplotlib scales by area by default.
    - Value of 1.94 of left, value of 0.89 on right, 700% scale
    ![Screenshot from 2020-09-09 21-59-56](https://user-images.githubusercontent.com/20176218/92683480-d4379100-f2e7-11ea-87ab-a2f84bb302c8.png)

2. Do we want the non-values to have the specific regions outlines (the R implementation, first screenshot), or is the state background sufficient (my current Python implementation, second screenshot)?
![Screenshot from 2020-09-09 22-01-01](https://user-images.githubusercontent.com/20176218/92683532-fb8e5e00-f2e7-11ea-9eed-bcff02f327ec.png)
![Screenshot from 2020-09-09 22-45-55](https://user-images.githubusercontent.com/20176218/92686433-7eb2b280-f2ee-11ea-859b-63000e4381e2.png)

3. @ryantibs One thing I noticed on the R code for these plots: the R discretization function seems to null any values < 0.1 (or whatever the lower bound happens to be), so they don't get a bubble. The region still gets un-greyed, so you know the region reported a value < 0.1 and wasn't missing. Is this intentional? I wasn't able to find the bubble plots on the covidcast website so didn't have reference there.
```
> breaks = c(0.1,0.2,0.3,0.4)
> dis_fun(c(0,0.05,0.1))
[1]  NA  NA 0.1
```
Empty values:

![Screenshot from 2020-09-09 20-17-07](https://user-images.githubusercontent.com/20176218/92687037-a5251d80-f2ef-11ea-854a-cc8fccb2ffc6.png)
![Screenshot from 2020-09-09 23-02-35](https://user-images.githubusercontent.com/20176218/92687562-a145cb00-f2f0-11ea-8aad-48d8d0adbfee.png)

Right now I have it so any values less than the second size get rolled up into the first one (and similarly for the largest values), but it wouldn't be hard to switch it over to the R scheme if that's what we want.

**Comment by capnrefsmmat** [comment]
Scaling by radius is almost certainly not intentional; we probably want to edit the R package to use [`scale_size()`](https://ggplot2.tidyverse.org/reference/scale_size.html) to ensure size = area.

I'm not sure about the other points, so I'll let Ryan answer based on how he decided the original function's behavior.

**Comment by capnrefsmmat** [comment]
Filed #73 to track that problem.

**Comment by ryantibs** [comment]
Thanks @chinandrew, great comments!  This has been a busy week and I haven't been able to get to this yet.  I'll hope to block off a bit of time, maybe sometime this weekend.  

1. Oops, I didn't intend to scale by radius, I'll take a look (and noted @capnrefsmmat, on #73).
2. I'm not sure which one I like better: yours looks cleaner, but truthfully I think both are pretty confusing; there's so much missingness here, it's just hard to interpret a bubble plot.  I just don't like bubble plots in this setting.  So I guess I'm fine with either one (but I'd suggest you make the border color less dark in your implementation, because the dark border around counties makes it hard read all the small counties in the east).
3. Hmmm, I can't honestly remember, I'll comment when I look back at the code.

**Comment by ryantibs** [comment]
@chinandrew @capnrefsmmat Sorry to be so late on this.  I finally managed to find a bit of time now to look at it.  With respect to Andrew's questions: 

1. So there's a couple things going on here.  First of all, the bubble sizes are *supposed* to be evenly-spaced across the range from `min_size` to `max_size`.  Here and throughout, when I say size, I mean area.  Note that this doesn't necessarily mean that a value of 4 should have a bubble size twice as big as a value of 2.  It all depends on what else is in the scale.  (To say it differently: the current implementation defines the scale to use a constant additive increase from one bubble size to the next, which seems visually reasonable.  When I tried defining the scale to use a constant multiplicative increase from one bubble size to the next, it looked much worse.  So I stuck with the original plan.)

That said, there still was something funny going on here ... and I'm glad we caught it.  I've been using `ggplot2::scale_size_manual()` to scale the bubble sizes, which actually sets the *radius*, and not the *area*.  This is very confusing, because `ggplot2::scale_size()` sets the *area* and not the *radius*.  So, I managed to fix that now, which should close #73.

(My fixes are sitting in a branch called "plotting-fixes"; I haven't submitted a PR yet since I haven't figured out that weird bubble-in-Canada issue #76).

3. This was intentional.  If you look at the documentation for `plot.covidcast_signal()`, in particular the description of the argument `num_bins`, you'll see that it explains that the bin for zero is special, and that values mapped to the zero bin aren't drawn.  I had to somewhat arbitrarily choose a boundary between the end of the zero bin and the next bin (unless the user specifies `breaks`, and then this is taken care of by their definition).  I chose `min(0.1, range[2] / num_bins)`.
